crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1047,6 +1053,13 @@ impl World {
         unsafe { data.with_type::<R>() }
     }
 
+    /// Gets a mutable reference to the resource of type `T` if it exists,
+    /// otherwise inserts the resource using the result of calling `T::default`.
+    #[inline]
+    pub fn get_resource_or_init<R: Resource + Default>(&mut self) -> Mut<'_, R> {
+        self.get_resource_or_insert_with(R::default)
```

Comment:
IMO this should use the `FromWorld` trait, for consistency with the other init_resource methods.